sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -764,7 +775,8 @@ private void recordRequestTimeline(
             CosmosMeterOptions timelineOptions = clientAccessor.getMeterOptions(
                 client,
                 name);
-            if (!timelineOptions.isEnabled()) {
+            if (!timelineOptions.isEnabled() ||
+                (timelineOptions.isDiagnosticThresholdsFilteringEnabled() && !ctx.isThresholdViolated())) {
```

Comment:
question: why not have a dedicated meter threshold config - would it be helpful especially for cases where operationStatusCode != requestStatusCode

Reply:
I don't think there is a need for it - default will be all metrics - filtering only when there is a large number of partitions - and in that case I don't see why you would want to use different thresholds for different meters. It makes it overly complicated.